internal/buffer/search.go

```diff
@@ -6,6 +6,40 @@ import (
 	"github.com/zyedidia/micro/v2/internal/util"
 )
 
+func padRegexp(r *regexp.Regexp) (*regexp.Regexp, *regexp.Regexp, *regexp.Regexp, *regexp.Regexp) {
+	rPadStart := regexp.MustCompile(".(?:"+r.String()+")")
+	rPadEnd := regexp.MustCompile("(?:"+r.String()+").")
+	rPadBoth := regexp.MustCompile(".(?:"+r.String()+").")
+	return r, rPadStart, rPadEnd, rPadBoth
+}
+
+func findLineParams(b *Buffer, start, end Loc, i int, rPadded [4]*regexp.Regexp) ([]byte, int, *regexp.Regexp) {
+	l := b.LineBytes(i)
+	charpos := 0
+	ri := 0 // rPadNone
+
+	if i == end.Y {
+		nchars := util.CharacterCount(l)
+		end.X = util.Clamp(end.X, 0, nchars)
+		if end.X < nchars {
+			l = util.SliceStart(l, end.X+1)
+			ri = 2 // rPadEnd
+		}
+	}
+
+	if i == start.Y {
+		nchars := util.CharacterCount(l)
+		start.X = util.Clamp(start.X, 0, nchars)
+		if start.X > 0 {
+			charpos = start.X-1
```

Comment:
gofmt wants it to be `start.X - 1` (with spaces).

Reply:
```suggestion
			charpos = start.X - 1
```